### ISSUE TITLE
Crash on WearOS

### ISSUE THREAD

**Issue by bigbigdaddy** [google/filament#4057]
I build samples on my android watch device and it goes black screen or crashes.
the watch is opengles3.0 and android8.1.0

**Comment by romainguy** [comment]
Android Wear devices are supported. Could you please describe the crash, provide logs, and describe the device you are using please?

**Comment by bigbigdaddy** [comment]
2021-06-03 21:47:50.497 18757-18757/? I/zygote: Late-enabling -Xcheck:jni
2021-06-03 21:47:50.907 18757-18757/com.google.android.filament.litcube I/Filament: FEngine (32 bits) created at 0x96080000 (threading is enabled)
2021-06-03 21:47:50.911 18757-18784/com.google.android.filament.litcube D/Filament: FEngine resolved backend: OpenGL
2021-06-03 21:47:50.913 18757-18784/com.google.android.filament.litcube I/Adreno-EGL: <qeglDrvAPI_eglInitialize:379>: QUALCOMM Build: 03/03/19, 8204ab1, Ibc8110c489
2021-06-03 21:47:50.927 18757-18784/com.google.android.filament.litcube I/zygote: android::hardware::configstore::V1_0::ISurfaceFlingerConfigs::hasWideColorDisplay retrieved: 0
2021-06-03 21:47:50.931 18757-18784/com.google.android.filament.litcube W/Adreno-EGL: <qeglDrvAPI_eglGetConfigAttrib:612>: EGL_BAD_ATTRIBUTE
2021-06-03 21:47:50.936 18757-18784/com.google.android.filament.litcube W/Adreno-ES20: <get_gpu_clk:229>: open failed: errno 13
2021-06-03 21:47:50.931 18757-18757/com.google.android.filament.litcube W/FEngine::loop: type=1400 audit(0.0:226): avc: denied { search } for name="kgsl-3d0" dev="sysfs" ino=18990 scontext=u:r:untrusted_app:s0:c512,c768 tcontext=u:object_r:sysfs_kgsl:s0 tclass=dir permissive=0
2021-06-03 21:47:51.080 18757-18787/com.google.android.filament.litcube D/OpenGLRenderer: HWUI GL Pipeline
2021-06-03 21:47:51.147 18757-18787/com.google.android.filament.litcube I/OpenGLRenderer: Initialized EGL, version 1.4
2021-06-03 21:47:51.147 18757-18787/com.google.android.filament.litcube D/OpenGLRenderer: Swap behavior 2
2021-06-03 21:47:51.150 18757-18787/com.google.android.filament.litcube W/Adreno-ES20: <get_gpu_clk:229>: open failed: errno 13
2021-06-03 21:47:51.141 18757-18757/com.google.android.filament.litcube W/RenderThread: type=1400 audit(0.0:227): avc: denied { search } for name="kgsl-3d0" dev="sysfs" ino=18990 scontext=u:r:untrusted_app:s0:c512,c768 tcontext=u:object_r:sysfs_kgsl:s0 tclass=dir permissive=0
2021-06-03 21:47:51.155 18757-18787/com.google.android.filament.litcube W/Adreno-EGL: <qeglDrvAPI_eglGetConfigAttrib:612>: EGL_BAD_ATTRIBUTE
2021-06-03 21:47:51.160 18757-18787/com.google.android.filament.litcube W/Adreno-EGL: <qeglDrvAPI_eglGetConfigAttrib:612>: EGL_BAD_ATTRIBUTE
2021-06-03 21:47:51.811 18757-18784/com.google.android.filament.litcube W/Adreno-EGL: <qeglDrvAPI_eglGetConfigAttrib:612>: EGL_BAD_ATTRIBUTE
    
    --------- beginning of crash
2021-06-03 21:47:51.912 18757-18784/com.google.android.filament.litcube A/libc: Fatal signal 11 (SIGSEGV), code 1, fault addr 0x60 in tid 18784 (FEngine::loop), pid 18757 (ilament.litcube)

**Comment by bigbigdaddy** [comment]
My device is OPPO Watch OW19W2
![image](https://user-images.githubusercontent.com/45582510/120656243-08766180-c4b6-11eb-89d1-0b1557c3c50d.png)

**Comment by romainguy** [comment]
Thanks, but you cut off the logs just at the beginning of the crash. Could you please show the rest of the logs so we know where it crashes?

**Comment by bigbigdaddy** [comment]
These are all the logs，look at this. maybe that's what caused it

2021-06-03 21:47:50.931 18757-18784/com.google.android.filament.litcube W/Adreno-EGL: <qeglDrvAPI_eglGetConfigAttrib:612>: EGL_BAD_ATTRIBUTE
2021-06-03 21:47:50.936 18757-18784/com.google.android.filament.litcube W/Adreno-ES20: <get_gpu_clk:229>: open failed: errno 13

**Comment by romainguy** [comment]
No, I'm interested in what's after:

```
2021-06-03 21:47:51.912 18757-18784/com.google.android.filament.litcube A/libc: Fatal signal 11 (SIGSEGV), code 1, fault addr 0x60 in tid 18784 (FEngine::loop), pid 18757 (ilament.litcube)
```

**Comment by bigbigdaddy** [comment]
--------- beginning of crash
06-04 14:33:42.159 20505 20523 F libc    : Fatal signal 11 (SIGSEGV), code 1, fault addr 0x60 in tid 20523 (FEngine::loop), pid 20505 (ilament.litcube)
06-04 14:33:42.275 20534 20534 I crash_dump32: obtaining output fd from tombstoned, type: kDebuggerdTombstone
06-04 14:33:42.276   554   554 I /system/bin/tombstoned: received crash request for pid 20505
06-04 14:33:42.278 20534 20534 I crash_dump32: performing dump of process 20505 (target tid = 20523)
06-04 14:33:42.278 20534 20534 F DEBUG   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
06-04 14:33:42.280 20534 20534 F DEBUG   : Build fingerprint: 'OPPO/OW19W2/OW19W2:8.1.0/OPM1.171019.011/1572606903:user/release-keys'
06-04 14:33:42.281 20534 20534 F DEBUG   : Revision: '0'
06-04 14:33:42.281 20534 20534 F DEBUG   : ABI: 'arm'
06-04 14:33:42.281 20534 20534 F DEBUG   : pid: 20505, tid: 20523, name: FEngine::loop  >>> com.google.android.filament.litcube <<<
06-04 14:33:42.281 20534 20534 F DEBUG   : signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x60
06-04 14:33:42.281 20534 20534 F DEBUG   : Cause: null pointer dereference
06-04 14:33:42.281 20534 20534 F DEBUG   :     r0 00000001  r1 a16dc371  r2 0000003c  r3 9407ceec
06-04 14:33:42.281 20534 20534 F DEBUG   :     r4 94a9e500  r5 00000000  r6 00000006  r7 95680310
06-04 14:33:42.281 20534 20534 F DEBUG   :     r8 94a9c904  r9 94a9ca64  sl 00000000  fp 94a88b4c
06-04 14:33:42.281 20534 20534 F DEBUG   :     ip 00000000  sp 95680140  lr a16eacbb  pc a16eacd2  cpsr 60070030
06-04 14:33:42.536 20534 20534 F DEBUG   : 
06-04 14:33:42.536 20534 20534 F DEBUG   : backtrace:
06-04 14:33:42.536 20534 20534 F DEBUG   :     #00 pc 00042cd2  /vendor/lib/libsc-a3xx.so (Compiler::Compile(ILProgram*)+31441)
06-04 14:33:42.536 20534 20534 F DEBUG   :     #01 pc 00047819  /vendor/lib/libsc-a3xx.so (Compiler::CompileShader(unsigned char*, unsigned char*, unsigned int const*, CompilerExternal*)+224)
06-04 14:33:42.536 20534 20534 F DEBUG   :     #02 pc 001cd379  /vendor/lib/libsc-a3xx.so (sclState::link(sclInputShaderPair const*, sclCompilerParams const&)+9944)
06-04 14:33:42.536 20534 20534 F DEBUG   :     #03 pc 001d2f7f  /vendor/lib/libsc-a3xx.so (__link_shaders+330)
06-04 14:33:42.536 20534 20534 F DEBUG   :     #04 pc 000960b5  /vendor/lib/egl/libGLESv2_adreno.so (core_glLinkProgram+1388)
06-04 14:33:42.537 20534 20534 F DEBUG   :     #05 pc 000dec0f  /data/app/com.google.android.filament.litcube-2784eCXz70fa8W4k11PyuA==/lib/arm/libfilament-jni.so

**Comment by romainguy** [comment]
Thank you. Unfortunately it's a crash and therefore a GPU driver bug. We can look for a workaround but we'd need the same device to be able to repro and find said workaround :(

**Comment by pixelflinger** [comment]
not actionable at this point

**Comment by pixelflinger** [comment]
This might be fixed by #5754
